src/data_race.rs

```diff
@@ -633,8 +635,15 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: MiriEvalContextExt<'mir, 'tcx> {
         use AtomicRwOp::*;
         let acquire = matches!(atomic, Acquire | AcqRel | SeqCst);
         let release = matches!(atomic, Release | AcqRel | SeqCst);
-        let this = self.eval_context_ref();
-        this.validate_atomic_op(place, atomic, "Atomic RMW", move |memory, clocks, index, _| {
+        let this = self.eval_context_mut();
+        this.validate_atomic_op(
+            place,
+            atomic,
+            "Atomic RMW",
+            // For yields the atomic write overrides all effects of the atomic read
+            // so it is treated as an atomic write.
+            true,
```

Comment:
Using `bool` here makes it hard to figure out what `true` and `false` even mean... would it make sense to introduce a 2-variant `enum` for this purpose?